### PR TITLE
pyload-ng: init at 0.5.0b3.dev72

### PR DESCRIPTION
**Pull request by ruby0b**
###### Description of changes
This packages the newer pyload-ng version of [pyload](https://pyload.net/), a download manager with support for 1-click-hosting sites.

The old pyload got dropped from nixpkgs in 2021 due to being unmaintained: https://github.com/NixOS/nixpkgs/commit/c0914a76c40230fb06eb6df8f54995796edef3ef

Potential future work: writing a NixOS service for autostarting

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ruby0b** [comment]
should the pyload throw alias be removed then (or mention the pyload-ng package)?

**Comment by SuperSandro2000** [comment]
we could do that

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -7625,6 +7627,8 @@ self: super: with self; {
 
   pylddwrap = callPackage ../development/python-modules/pylddwrap { };
 
+  pyload-ng = callPackage ../development/python-modules/pyload-ng { };
```
[the remark this answers is not in the document]
Comment (ruby0b) [reply pkgs/top-level/python-packages.nix:7630]: makes sense, it isn't really usable as a library anyway I think

**Comment by SuperSandro2000** [comment]
@ofborg eval

File: pkgs/top-level/aliases.nix [hunk 2]
```diff
@@ -1380,7 +1380,7 @@ mapAliases ({
   pybitmessage = throw "pybitmessage was removed from nixpkgs as it is stuck on python2"; # Added 2022-01-01
   pyext = throw "pyext was removed because it does not support python 3.11, is upstream unmaintained and was unused"; # Added 2022-11-21
   pygmentex = texlive.bin.pygmentex; # Added 2019-12-15
-  pyload = throw "pyload has been removed from nixpkgs, as it was unmaintained"; # Added 2021-03-21
+  pyload = throw "pyload has been removed from nixpkgs. Use pyload-ng instead."; # Edited to mention pyload-ng 2023-05-24
```
Comment (SuperSandro2000) [inline pkgs/top-level/aliases.nix:1383]:
```suggestion
  pyload = throw "pyload has been removed from nixpkgs. Use pyload-ng instead."; # Added 2021-03-21
```
the comment is just that we know when we can remove the alias

File: pkgs/applications/networking/pyload-ng/default.nix [hunk 3]
```diff
@@ -0,0 +1,56 @@
+{ lib, python3 }:
+
+python3.pkgs.buildPythonApplication rec {
+  version = "0.5.0b3.dev69";
+  pname = "pyload-ng";
+  format = "pyproject";
+
+  disabled = python3.pkgs.pythonOlder "3.8";
+
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/pyload-ng/default.nix:9]:
```suggestion
```
python3 will never be older than 3.8

File: pkgs/applications/networking/pyload-ng/default.nix [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  src = python3.pkgs.fetchPypi {
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/pyload-ng/default.nix:10]:
```suggestion
  src = python3.fetchPypi {
```
that was recently deprecated

**Review by SuperSandro2000 — COMMENTED** [review]
two minor things, otherwise LGTM

File: pkgs/applications/networking/pyload-ng/default.nix (hunk 4 above)
Comment (ruby0b) [reply pkgs/applications/networking/pyload-ng/default.nix:8]: python3.fetchPypi doesn't seem to exist

Comment (SuperSandro2000) [reply pkgs/applications/networking/pyload-ng/default.nix:8]: probably pkgs.fetchPypi https://github.com/NixOS/nixpkgs/pull/234043

```suggestion
  src = fetchPypi {
```

File: pkgs/applications/networking/pyload-ng/default.nix [hunk 5]
```diff
@@ -0,0 +1,54 @@
+{ lib, python3 }:
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/pyload-ng/default.nix:1]:
```suggestion
{ lib, fetchPypi, python3 }:
```

**Review by SuperSandro2000 — COMMENTED** [review]
@ofborg eval

**Comment by ruby0b** [comment]
should be ready now :)

**Review by SuperSandro2000 — COMMENTED** [review]
please rebase otherwise LGTM

**Comment by ruby0b** [comment]
oh no! what happened? I did the rebase on a fresh clone but afterwards it required me to pull from the remote pyload and merge it with my local branch...

**Comment by ruby0b** [comment]
Oh, I had to force push since the histories had diverged, makes sense in retrospect. Sorry for pinging like 50 people :')

**Comment by Ma27** [comment]
Please file a new PR: we cannot unsubcribe everyone, so all the activity in here will ping a lot of people.

**Pull request closed without merging**